src/base/utils/fs.cpp

```diff
@@ -234,120 +180,65 @@ bool Utils::Fs::sameFiles(const QString &path1, const QString &path2)
-    {allowSeparators
-        ? QLatin1String("[\\0:]")
-        : QLatin1String("[\\0/:]")};
-#else
-    const QRegularExpression regex
-    {allowSeparators
-        ? QLatin1String("[\\0]")
-        : QLatin1String("[\\0/]")};
-#endif
-    return !name.contains(regex);
-}
+    const QRegularExpression regex {QLatin1String("[:?\"*<>|]+")};
 
-qint64 Utils::Fs::freeDiskSpaceOnPath(const QString &path)
-{
-    return QStorageInfo(path).bytesAvailable();
-}
+    QString validPathStr = name;
+    validPathStr.replace(regex, pad);
 
-QString Utils::Fs::branchPath(const QString &filePath, QString *removed)
-{
-    QString ret = toUniformPath(filePath);
-    if (ret.endsWith('/'))
-        ret.chop(1);
-    const int slashIndex = ret.lastIndexOf('/');
-    if (slashIndex >= 0)
-    {
-        if (removed)
-            *removed = ret.mid(slashIndex + 1);
-        ret = ret.left(slashIndex);
-    }
-    return ret;
+    return Path(validPathStr);
 }
 
-bool Utils::Fs::sameFileNames(const QString &first, const QString &second)
+qint64 Utils::Fs::freeDiskSpaceOnPath(const Path &path)
 {
-#if defined(Q_OS_UNIX) || defined(Q_WS_QWS)
-    return QString::compare(first, second, Qt::CaseSensitive) == 0;
-#else
-    return QString::compare(first, second, Qt::CaseInsensitive) == 0;
-#endif
+    return QStorageInfo(path.data()).bytesAvailable();
 }
 
-QString Utils::Fs::expandPath(const QString &path)
+Path Utils::Fs::tempPath()
 {
-    const QString ret = path.trimmed();
-    if (ret.isEmpty())
-        return ret;
-
-    return QDir::cleanPath(ret);
-}
-
-QString Utils::Fs::expandPathAbs(const QString &path)
-{
-    return QDir(expandPath(path)).absolutePath();
-}
-
-QString Utils::Fs::tempPath()
-{
-    static const QString path = QDir::tempPath() + "/.qBittorrent/";
-    QDir().mkdir(path);
+    static const Path path = Path(QDir::tempPath()) / Path(".qBittorrent");
+    mkdir(path);
     return path;
 }
 
-bool Utils::Fs::isRegularFile(const QString &path)
+bool Utils::Fs::isRegularFile(const Path &path)
 {
     struct ::stat st;
-    if (::stat(path.toUtf8().constData(), &st) != 0)
+    if (::stat(path.toString().toUtf8().constData(), &st) != 0)
     {
         //  analyse erno and log the error
         const auto err = errno;
         qDebug("Could not get file stats for path '%s'. Error: %s"
-               , qUtf8Printable(path), qUtf8Printable(strerror(err)));
+               , qUtf8Printable(path.toString()), qUtf8Printable(strerror(err)));
         return false;
     }
 
     return (st.st_mode & S_IFMT) == S_IFREG;
 }
 
-#if !defined Q_OS_HAIKU
-bool Utils::Fs::isNetworkFileSystem(const QString &path)
+bool Utils::Fs::isNetworkFileSystem(const Path &path)
 {
-#if defined(Q_OS_WIN)
-    const std::wstring pathW {path.toStdWString()};
-    auto volumePath = std::make_unique<wchar_t[]>(path.length() + 1);
-    if (!::GetVolumePathNameW(pathW.c_str(), volumePath.get(), (path.length() + 1)))
+#if defined Q_OS_HAIKU
+    return false;
```

Comment:
It makes code cleaner, IMO.
Why all caller code should care of what platform is supported currently? It should implement its (upper level) logic. If, for example, support of detecting network file systems on Haiku is added, it will affect only the place that directly depends on it.